build/setup.pm

```diff
@@ -437,6 +459,7 @@ our %OS_LINUX = (
 our %OS_OPENBSD = (
     %OS_POSIX,
 
+    cc => (qx!egcc -v 2>&1 >$devnull! !~ 'egcc') ? 'gcc' : 'egcc',
     syslibs => [ @{$OS_POSIX{syslibs}}, qw( kvm ) ],
```

Comment:
This would get executed everywhere, even on non-BSDs, won't it? For example, here's what that `qx` gives on Windows:

```
C:\rakudo>perl -e "qx!egcc -v 2>&1 >$devnull! !~ 'egcc'"
The syntax of the command is incorrect.
```

Perhaps, we should make an anon sub `cc => sub { (qx!egcc -v 2>&1 >$devnull! !~ 'egcc') ? 'gcc' : 'egcc' },` and then on line 548 change `$OS_OPENBSD{cc}` to `$OS_OPENBSD{cc}()`

Reply:
I run into the same problem on my machine as well. I'm not sure if the fix you gave worked though:

```
kennel% perl -e "sub a() { (qx!cc -v 2>&1 >$devnull! 'gcc') ? 'gcc' : 'clang' }; print a()"
zsh: event not found: cc
```

Reply:
How well would this work?
```perl
cc => system('egcc -v > /dev/null 2>&1') ? 'gcc' : 'egcc',
```

Reply:
> I'm not sure if the fix you gave works either

Windows has reversed quotes on the command line, which is why I had `-e ""`  with single quotes for strings. If you're running that command on non-Windows, you need to flip the quotes (also, you don't have `$devnull` variable declared):

    $ perl -e 'sub a() { (qx!cc -v 2>&1 >/dev/null! !~ "gcc") ? "gcc" : "clang" }; print a()'
    clang